src/cc65/coptmisc.c

```diff
@@ -555,6 +556,65 @@ unsigned OptAXLoad (CodeSeg* S)
 
 
 
+unsigned OptAXLoad2 (CodeSeg* S)
+/* Merge ldy/jsr incaxy/jsr ldaxi into ldy/jsr ldaxidx */
+{
+    unsigned Changes = 0;
+    unsigned I;
+
+    /* Walk over the entries */
+    I = 0;
+    while (I < CS_GetEntryCount (S)) {
+
+        signed Val;
+        CodeEntry* E[3];
+        CodeEntry *X;
+        char *End;
+
+        /* Get the next entry */
+        E[0] = CS_GetEntry (S, I);
+
+        /* Check for ldy followed by incaxy followed by jsr/jmp ldaxi */
+        if (E[0]->OPC == OP65_LDY                            &&
+            E[0]->AM == AM65_IMM                             &&
```

Comment:
```suggestion
        if (E[0]->OPC == OP65_LDY                            &&
            CE_IsConstImm (E[0])                             &&
```
The reason why this is required is that the argument to "ldy immediate" is not necessarily numeric. The following translates to the immediate load of a symbolic constant into Y:

```
void f(void)
{
    asm ("ldy #<_f");
    asm ("sty _a");
}
```
Just look at the generated assembler code. It would match your test but converting it to a numerical value will silently fail.

Reply:
Thanks! I wasn't aware of that :)